There are test failures that look related:
```
[ERROR] Failures: 
[ERROR]   OpenApiTestCase.testOpenAPIJSON:65 Normal and RX/Single have same schema ==> expected: <{"get":{"responses":{"200":{"description":"OK","content":{"*/*":{"schema":{"$ref":"#/components/schemas/String"}}}}}}}> but was: <{"get":{"responses":{"200":{"description":"OK","content":{"*/*":{"schema":{"type":"string"}}}}}}}>
[INFO] 
```

--- a/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
+++ b/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java
@@ -23,11 +23,13 @@ import org.eclipse.microprofile.openapi.models.OpenAPI;
 import org.jboss.jandex.AnnotationInstance;
 import org.jboss.jandex.AnnotationTarget;
 import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.CompositeIndex;
 import org.jboss.jandex.DotName;
 import org.jboss.jandex.IndexView;
 import org.jboss.jandex.Type;
 
 import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanArchiveIndexBuildItem;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.ApplicationArchivesBuildItem;
@@ -96,8 +98,9 @@ public class SmallRyeOpenApiProcessor {
 
     @BuildStep
     public void registerOpenApiSchemaClassesForReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
-            BuildProducer<ReflectiveHierarchyBuildItem> reflectiveHierarchy, CombinedIndexBuildItem combinedIndexBuildItem) {
-        IndexView index = combinedIndexBuildItem.getIndex();
+            BuildProducer<ReflectiveHierarchyBuildItem> reflectiveHierarchy, CombinedIndexBuildItem combinedIndexBuildItem,
+            BeanArchiveIndexBuildItem beanArchiveIndexBuildItem) {
+        IndexView index = CompositeIndex.create(combinedIndexBuildItem.getIndex(), beanArchiveIndexBuildItem.getIndex());
 
         // Generate reflection declaration from MP OpenAPI Schema definition
         // They are needed for serialization.
@@ -180,12 +183,16 @@ public class SmallRyeOpenApiProcessor {
     public void build(ApplicationArchivesBuildItem archivesBuildItem,
             CombinedIndexBuildItem combinedIndexBuildItem, BuildProducer<FeatureBuildItem> feature,
             Optional<ResteasyJaxrsConfigBuildItem> resteasyJaxrsConfig,
-            BuildProducer<GeneratedWebResourceBuildItem> resourceBuildItemBuildProducer) throws Exception {
+            BuildProducer<GeneratedWebResourceBuildItem> resourceBuildItemBuildProducer,
+            BeanArchiveIndexBuildItem beanArchiveIndexBuildItem) throws Exception {
+
+        IndexView index = CompositeIndex.create(combinedIndexBuildItem.getIndex(), beanArchiveIndexBuildItem.getIndex());
+
         feature.produce(new FeatureBuildItem(FeatureBuildItem.SMALLRYE_OPENAPI));
         OpenAPI staticModel = generateStaticModel(archivesBuildItem);
         OpenAPI annotationModel;
         if (resteasyJaxrsConfig.isPresent()) {
-            annotationModel = generateAnnotationModel(combinedIndexBuildItem.getIndex(), resteasyJaxrsConfig.get());
+            annotationModel = generateAnnotationModel(index, resteasyJaxrsConfig.get());
         } else {
             annotationModel = null;
         }

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/OpenApiTestCase.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/OpenApiTestCase.java
@@ -61,13 +61,41 @@ public class OpenApiTestCase {
 
         // test RESTEasy extensions
 
+        JsonObject schemasObj = obj.getJsonObject("components").getJsonObject("schemas");
+        String testSchemaType = schemaType("200", "*/*", testObj.getJsonObject("get").getJsonObject("responses"), schemasObj);
+        String rxSchemaType = schemaType("200", "*/*", injectionObj.getJsonObject("get").getJsonObject("responses"),
+                schemasObj);
         // make sure String, CompletionStage<String> and Single<String> are detected the same
-        Assertions.assertEquals(testObj, injectionObj, "Normal and RX/Single have same schema");
+        Assertions.assertEquals(testSchemaType,
+                rxSchemaType,
+                "Normal and RX/Single have same schema");
         JsonObject csObj = paths.getJsonObject("/test/cs");
-        Assertions.assertEquals(testObj, csObj, "Normal and RX/CS have same schema");
+        Assertions.assertEquals(testSchemaType,
+                schemaType("200", "*/*", csObj.getJsonObject("get").getJsonObject("responses"), schemasObj),
+                "Normal and RX/CS have same schema");
 
         JsonObject paramsObj = paths.getJsonObject("/test/params/{path}");
         JsonObject params2Obj = paths.getJsonObject("/test/params2/{path}");
         Assertions.assertEquals(paramsObj, params2Obj, "Normal and RESTEasy annotations have same schema");
+    }
+
+    protected static String schemaType(String responseCode, String mediaType, JsonObject responses, JsonObject schemas) {
+        JsonObject responseContent = responses.getJsonObject(responseCode).getJsonObject("content");
+        if (responseContent == null) {
+            return null;
+        }
+        JsonObject schemaObj = responseContent.getJsonObject(mediaType)
+                .getJsonObject("schema");
+
+        if (schemaObj.containsKey("type")) {
+            return schemaObj.getString("type");
+        } else if (schemaObj.containsKey("$ref")) {
+            String schemaReference = schemaObj.getString("$ref");
+            String schemaRefType = schemaReference.substring(schemaReference.lastIndexOf("/") + 1);
+            return schemas.getJsonObject(schemaRefType).getString("type");
+        }
+
+        throw new IllegalArgumentException(
+                "Cannot retrieve schema type for response " + responseCode + " and media type " + mediaType);
     }
 }

